authors/apps/authentication/views.py

```diff
@@ -276,3 +276,129 @@ def create(self, request):
+            }
+            return Response(message, status=status.HTTP_406_NOT_ACCEPTABLE)
+        # import pdb; pdb.set_trace()
+
+        try:
+            already_followed = follower.following.filter(pk=id).exists()
+            if already_followed:
+                return Response({
+                    'error': 'You already follow this user'
+                },
+                    status=status.HTTP_406_NOT_ACCEPTABLE)
+        except Exception as e:
+            return Response({
+                'error': 'Error when following',
+                'message': str(e)
+            },
+                status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+        follower.following.add(to_be_followed)
+        follower.save()
+
+        serializer = FollowUnfollowSerializer(to_be_followed)
+        message = {
+            "message": "Profile successfully followed",
+            "user": serializer.data
+        }
+        return Response(message, status=status.HTTP_200_OK)
+
+    def delete(self, request, id, format=None):
+        """
+        Deletes a follow relationship between the user sending the
+        request and the user with the username passed
+        """
+        try:
+            to_be_unfollowed = User.objects.get(pk=id)
+        except Exception:
+            return Response({
+                'error': "User not found"
+            }, status=status.HTTP_404_NOT_FOUND)
+
+        follower = User.objects.get(pk=request.user.id)
+
+        relationship = follower.following.filter(pk=id).exists()
+        if not relationship:
+            return Response(
+                {
+                    'error': 'You do not follow this user'
+                },
+                status=status.HTTP_400_BAD_REQUEST
+            )
+
+        follower.following.remove(id)
+        serializer = FollowUnfollowSerializer(to_be_unfollowed)
+        message = {
+            "message": "Profile successfully unfollowed",
+            "user": serializer.data
+        }
+        return Response(message, status=status.HTTP_200_OK)
+
+
+class FollowerFollowingAPIView(generics.ListAPIView):
+    """
+    This API returns a list of user followers and following
+    """
+
+    def get_queryset(self):
+        user = get_object_or_404(User, id=self.kwargs['id'])
+
+        followed_friends = user.following.all()
+        following_friends = user.followers.all()
+
+        return {
+            "followed": followed_friends,
+            "followers": following_friends
+        }
+
+    def get(self, request, id, format=None):
+        """Returns the user's followed user"""
+
+        if self.get_queryset() is not None:
+
+            followed_friend_objects = self.get_queryset()["followed"]
+            following_friend_objects = self.get_queryset()["followers"]
+
+            users_followed = [u for u in followed_friend_objects]
+            users_following = [u for u in following_friend_objects]
+
+            follower_serializer = FollowerFollowingSerializer(
```

Comment:
You can pass in the `querysets` from `get_queryset` with objects you don't have to create other lists.

Reply:
Done